### PR TITLE
Traffic label transparent transmission plug-in: adapt to rocketmq5.x version

### PR DESCRIPTION
**Pull request by daizhenyu**
【Fix issue】#1244

[Modification content] The tag-transmission-common module is added to provide config and abstract interceptor capabilities for the traffic tag transparent transmission plug-in; the tag-transmission-rocketmq5.x-plugin is added to adapt the plug-in to the rocketmq5.x version; modified The tag-transmission-plugin module is installed, and the common logic of the two tag transparent transmission plug-ins is extracted into the tag-transmission-common module.

[Use case description] Not required

[Self-test situation] 1. Local static check and cleanup passed

[Scope of Impact] User documentation needs to be updated

### CODE REVIEW

File: sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqConsumerInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    private static final String ROCKETMQ_SELECT_CLASSNAME = "org.apache.rocketmq.common.message.Message";
+
+    /**
+     * getBody拦截方法应被过滤的前缀
+     */
+    private static final String ROCKETMQ_FILER_PREFIX = "org.apache.rocketmq";
+
+    @Override
+    public ExecuteContext doBefore(ExecuteContext context) {
+        StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
```
Comment (lilai23) [inline sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqConsumerInterceptor.java:56]: 拉最新代码和4.x版本一样，判断是否是生产线程

Comment (daizhenyu) [reply sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqConsumerInterceptor.java:56]: 已修改

**Review by lilai23 — DISMISSED** [review]

File: sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                newProperties.append(key);
+                newProperties.append(LINK_MARK);
+                newProperties.append((String) null);
+                newProperties.append(SPLIT_MARK);
+                continue;
+            }
+            newProperties.append(key);
+            newProperties.append(LINK_MARK);
+            newProperties.append(values.get(0));
+            newProperties.append(SPLIT_MARK);
```
Comment (luanwenfei-venus) [inline sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java]: 建议这两块append，部分代码复用

File: sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        }
+        if (newProperties.length() == 0) {
+            return oldProperties;
+        }
+        if (oldProperties == null || oldProperties.length() == 0) {
```
Comment (luanwenfei-venus) [inline sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java:92]: 这里的两段删除char和拼接有些不明确，建议抽成方法或者添加注释来提升这里的可读性

File: sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java (hunk 2 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java]: 已修改

File: sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java (hunk 3 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-tag-transmission/tag-transmission-rocketmq5.x-plugin/src/main/java/com/huaweicloud/sermant/tag/transmission/rocketmqv5/interceptor/RocketmqProducerSendInterceptor.java:92]: 已修改

**Review by luanwenfei-venus — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**